### DISCUSSION TITLE
Sweep

### DISCUSSION THREAD

**Discussion by jgomez720** [KittyCAD/modeling-app#498] [New Feature Requests]
### Description
A tool that enables the user to extrude a sketch along a path

### Functionality
1. The user creates a path (usually this is a spline)
2. The user creates a sketch they want to extrude along that path. Most of the time, the path must start perpendicular to the sketch plane.

### Supporting Docs
https://github.com/KittyCAD/modeling-app/assets/114548659/51d7fc76-2c08-4912-a5fb-5f0529a192b6

**Example of sweep failure:**
![Screenshot 2023-11-08 at 9 30 04 AM](https://github.com/KittyCAD/modeling-app/assets/114548659/38709081-a394-44b2-a4cc-532194b5aa7e)
This occurs when there are bends that are too large to physically render.

## PseudoKCL
```
const sweepSketch = startSketchOn('XY')
  |> startProfileAt([2, 0], %)
  |> arc({
       angle_end: 360,
       angle_start: 0,
       radius: 2
     }, %)

const sweepPath = startSketchOn('YZ')
  |> startProfileAt([0.05, 0.05], %)
  |> line([0, 7], %)
  |> tangentialArc({
    offset: 90,
    radius: 5
  }, %)
  |> line([-3, 0], %)
  |> tangentialArc({
    offset: -90,
    radius: 5
  }, %)
  |> line([0, 7], %)

sweep({
  sketch: sweepSketch,
  path: sweepPath,
  remove: false // optional. Always add material, user should specify remove: true if they want it to remove material instead.
})
```
**Expected Result:**
![Screenshot 2023-11-08 at 9 56 21 AM](https://github.com/KittyCAD/modeling-app/assets/114548659/a984f465-94b2-49f1-a9fa-321c4cdf0b5a)
![Screenshot 2023-11-08 at 9 56 42 AM](https://github.com/KittyCAD/modeling-app/assets/114548659/3d776978-4f48-4455-b5ac-bd3020cda875)

Most of this I could do in KCL today, just not the sweep function.
![Screenshot 2023-11-08 at 9 50 48 AM](https://github.com/KittyCAD/modeling-app/assets/114548659/76a550ee-9506-431a-856c-968e54a00d4a)

**Comment by mlfarrell** [comment]
started some research on this one last week at https://github.com/KittyCAD/engine/issues/2128
